### PR TITLE
solution

### PR DESCRIPTION
**Pull request by liliya-dev**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -65,7 +67,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const interimStates = JSON.parse(JSON.stringify(state));
+  const statesArray = [];
+
+  for (const property of transforms) {
+    if (property['operation'] === 'addProperties') {
```
Comment (V2473) [inline src/transformStateWithClones.js:74]: use switch

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -65,7 +67,33 @@
...
+      for (const [key, value] of Object.entries(property['properties'])) {
+        interimStates[key] = value;
+      }
+      statesArray.push(JSON.parse(JSON.stringify(interimStates)));
```
Comment (V2473) [inline src/transformStateWithClones.js:78]: maybe use { ... interimStates }

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -65,7 +67,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const interimStates = JSON.parse(JSON.stringify(state));
```
Comment (V2473) [inline src/transformStateWithClones.js:70]: maybe use spread operator?

File: src/transformStateWithClones.js (hunk 1 above)
Comment (solaryasha) [inline src/transformStateWithClones.js:74]: For me it's okay, but Dima will ask you to use `switch` statement

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -65,7 +67,33 @@
...
+        delete interimStates[key];
+      }
+      statesArray.push(JSON.parse(JSON.stringify(interimStates)));
+    }
+
+    if (property.operation === 'removeProperties') {
+      for (const keys of property['properties']) {
+        delete interimStates[keys];
+      }
+      statesArray.push(JSON.parse(JSON.stringify(interimStates)));
```
Comment (solaryasha) [inline src/transformStateWithClones.js:92]: You repeated this block of code three times.
Better to write it one time after the switch loop.

File: src/transformStateWithClones.js [hunk 5]
```diff
@@ -65,7 +67,33 @@
...
+    }
```
Comment (solaryasha) [inline src/transformStateWithClones.js:93]: Here

File: src/transformStateWithClones.js [hunk 6]
```diff
@@ -65,7 +67,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const interimStates = JSON.parse(JSON.stringify(state));
```
Comment (danheim) [inline src/transformStateWithClones.js:70]: https://learn.javascript.ru/destructuring

File: src/transformStateWithClones.js [hunk 7]
```diff
@@ -65,7 +67,36 @@
...
+      }
+
+      case 'removeProperties': {
+        for (const keys of property['properties']) {
+          delete interimStates[keys];
+        }
+        break;
+      }
+    }
+    statesArray.push(JSON.parse(JSON.stringify(interimStates)));
```
Comment (danheim) [inline src/transformStateWithClones.js:96]: https://learn.javascript.ru/destructuring

**Pull request closed without merging**